primedev/client/clientauthhooks.cpp

```diff
@@ -17,52 +17,21 @@ AUTOHOOK(AuthWithStryder, engine.dll + 0x1843A0,
 void, __fastcall, (void* a1))
 // clang-format on
 {
-	// don't attempt to do Atlas auth if we are in vanilla compatibility mode
-	// this prevents users from joining untrustworthy servers (unless they use a concommand or something)
-	if (g_pVanillaCompatibility->GetVanillaCompatibility())
-	{
-		AuthWithStryder(a1);
-		return;
-	}
-
-	// game will call this forever, until it gets a valid auth key
-	// so, we need to manually invalidate our key until we're authed with northstar, then we'll allow game to auth with stryder
 	if (!g_pMasterServerManager->m_bOriginAuthWithMasterServerDone && Cvar_ns_has_agreed_to_send_token->GetInt() != DISAGREED_TO_SEND_TOKEN)
 	{
 		// if player has agreed to send token and we aren't already authing, try to auth
 		if (Cvar_ns_has_agreed_to_send_token->GetInt() == AGREED_TO_SEND_TOKEN &&
 			!g_pMasterServerManager->m_bOriginAuthWithMasterServerInProgress)
 			g_pMasterServerManager->AuthenticateOriginWithMasterServer(g_pLocalPlayerUserID, g_pLocalPlayerOriginToken);
-
-		// invalidate key so auth will fail
-		*g_pLocalPlayerOriginToken = 0;
 	}
 
 	AuthWithStryder(a1);
 }
 
-char* p3PToken;
-
-// clang-format off
-AUTOHOOK(Auth3PToken, engine.dll + 0x183760,
-char*, __fastcall, ())
-// clang-format on
-{
-	if (!g_pVanillaCompatibility->GetVanillaCompatibility() && g_pMasterServerManager->m_sOwnClientAuthToken[0])
-	{
-		memset(p3PToken, 0x0, 1024);
-		strcpy(p3PToken, "Protocol 3: Protect the Pilot");
-	}
-
-	return Auth3PToken();
-}
-
```

Comment:
Wanna explain why you decided to remove the protection against leaking Origin tokens?

Reply:
The 3p token protection stuff sort of nuked it which isn't helpful if you need to auth to vanilla servers after connecting to northstar, at least I think so anyway, maybe you could hook up engine.dll + 0x1841F0 but I don't know if that's for 3p or not. I can try it later if it matters that much at least

Reply:
Actually I am stupid it's ok ignore the above

Reply:
this is a really dumb "patch" any way you put it since this will cause you to DoS Stryder trying to get a valid 3p auth token (if you get 429, that's why) - the proper fix is to hook bf_write::WriteString and check retaddr, and if we just wrote serverfilter with a non-empty value and we're writing 3p token, then write a blank